pkg/assets/create/creater.go

```diff
@@ -0,0 +1,230 @@
+var newClientsFn = newClients
+
+func newClients(config *rest.Config) (dynamic.Interface, *discovery.DiscoveryClient, error) {
+	client, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// TODO: We can use cacheddiscovery.NewMemCacheClient(dc) and then call .Invalidate() instead of fetchLatestDiscoveryInfo.
+	// It will require more work in unit test though.
+	dc, err := discovery.NewDiscoveryClientForConfig(config)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return client, dc, nil
+}
+
+// allow to override in unit test
+var fetchLatestDiscoveryInfoFn = fetchLatestDiscoveryInfo
+
+func fetchLatestDiscoveryInfo(dc *discovery.DiscoveryClient) (meta.RESTMapper, error) {
+	gr, err := restmapper.GetAPIGroupResources(dc)
+	if err != nil {
+		return nil, err
+	}
+	return restmapper.NewDiscoveryRESTMapper(gr), nil
+}
+
+// create will attempt to create all manifests provided using dynamic client.
+// It will mutate the manifests argument in case the create succeeded for given manifest. When all manifests are successfully created the resulting
+// manifests argument should be empty.
+func create(manifests map[string]*unstructured.Unstructured, client dynamic.Interface, mapper meta.RESTMapper, options CreateOptions) (error, bool) {
+	// Sort all manifests, so in case they use number prefixes, we follow their order, which will increase
+	// the chances for the create loop to succeed faster.
+	sortedManifestPaths := []string{}
+	for key := range manifests {
+		sortedManifestPaths = append(sortedManifestPaths, key)
+	}
+	sort.Strings(sortedManifestPaths)
+
+	// Record all errors for the given manifest path (so when we report errors, users can see what manifest failed).
+	errs := map[string]error{}
+
+	// In case we fail to find a rest-mapping for the resource, force to fetch the updated discovery on next run.
+	reloadDiscovery := false
+
+	for _, path := range sortedManifestPaths {
+		gvk := manifests[path].GetObjectKind().GroupVersionKind()
+		mappings, err := mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+		if err != nil {
+			errs[path] = fmt.Errorf("unable to get REST mapping: %v", err)
+			reloadDiscovery = true
+			continue
+		}
+
+		if options.Verbose {
+			fmt.Fprintf(options.StdErr, "Creating %s ...\n", mappings.Resource.String())
+		}
+		if mappings.Scope.Name() == meta.RESTScopeNameRoot {
+			_, err = client.Resource(mappings.Resource).Create(manifests[path], metav1.CreateOptions{})
+		} else {
+			_, err = client.Resource(mappings.Resource).Namespace(manifests[path].GetNamespace()).Create(manifests[path], metav1.CreateOptions{})
+		}
+
+		// Resource already exists means we already succeeded
+		// This should never happen as we remove already created items from the manifest list, unless the resource existed beforehand.
+		if kerrors.IsAlreadyExists(err) {
+			delete(manifests, path)
+			continue
+		}
+
+		if err != nil {
+			errs[path] = fmt.Errorf("failed to create: %v", err)
+			continue
```

Comment:
so on error we stop the current loop. Should this be configurable? It might be that some (alphabetically) later resources are necessary for earlier resources to succeed. In CVO we probably want `continue`, in unsorted manifests we don't (not sure which case the rendered manifests are).

Reply:
oops, no. it's continue